### PR TITLE
Spells/Priest: Fix Power Infusion, shouldn't stack with Arcane Power (Mage)

### LINKED ISSUE #23614: [3.3.5] DB/Stacks: Priest: Power Infusion stack with Mage: Arcane Power
**Description:**

https://www.mmo-champion.com/threads/692845-Arcane-Power-Power-Infusion-Fail
https://wow.gamepedia.com/Power_Infusion
>Patch 1.10.2 (2006-05-02): This ability will no longer stack with  [Arcane Power]. If you gain both at the same time, Arcane Power will take precedence. 

ID - 10060 Power Infusion
ID - 12042 Arcane Power

Old Issue: #8705

**Possible Fix:**
(Updated here: #23619)
```sql
SET @GROUP_ID := 1122; -- Change with the proper ID if this is in use
DELETE FROM `spell_group` WHERE `id`=@GROUP_ID AND `spell_id` IN (10060,12042);
INSERT INTO `spell_group` (`id`,`spell_id`) VALUES
(@GROUP_ID,10060), -- Priest - Power Infusion
(@GROUP_ID,12042); -- Mage - Arcane Power

DELETE FROM `spell_group_stack_rules` WHERE `group_id`=@GROUP_ID;
INSERT INTO `spell_group_stack_rules` (`group_id`,`stack_rule`) VALUES
(@GROUP_ID,4);
```

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

https://github.com/TrinityCore/TrinityCore/commit/8a174d22957d3f641b4a6750cdc2a4ae88a88402

### PR DESCRIPTION
**Pull request by Jildor**
**Changes proposed:**

https://www.mmo-champion.com/threads/692845-Arcane-Power-Power-Infusion-Fail
https://wow.gamepedia.com/Power_Infusion
ID - 10060 Power Infusion
ID - 12042 Arcane Power

- Patch 1.10.2 (2006-05-02): This ability will no longer stack with [Arcane Power]. If you gain both at the same time, Arcane Power will take precedence.

- If you have Power Infusion Aura and cast Arcane Power, Power Infusion should be replaced by Arcane Power, but if you have Arcane Power and try to cast Power Infusion, you should get msg "A more powerful spell is active".

- Arcane Power and Power Infusion don't have the same effect, so stack_rule 4 (SPELL_GROUP_STACK_RULE_EXCLUSIVE_HIGHEST) isn't valid, so I put stack_rule 1 (SPELL_GROUP_STACK_RULE_EXCLUSIVE) for prevent stack the two auras and SpellScript for get error msg when you cast Power Infusion when target have Arcane Power Aura.

I don't see any other group of spells that need the same result without same effect, so can we do a exception?
In case of consider this a bit hacky, we need support in spell_group for this specific case.

**Target branch(es):** 3.3.5/master

- [X] 3.3.5
- [ ] master

**Issues addressed:** Closes #23614

**Tests performed:** tested in-game

### CODE REVIEW

**Comment by illfated** [comment]
For future PRs with a combination of SQL and C++ like this, with that amount of explanatory text in the SQL, I would like to recommend putting most of the comment text in the commit message (2 lines below the commit title) instead of in the SQL file (just a suggestion, in case you may find it to be useful information). I presume it will be too much to ask to change that in this PR, but I hope you can keep it in mind for any future PRs to avoid lots of comment text in the SQL file.

**Comment by Jildor** [comment]
Ok thx 😄 
Well, I'm going to remove unnecessary comments, it's explained in the main comment of PR ;)

File: sql/updates/world/3.3.5/9999_99_99_99_world_335.sql [hunk 1]
```diff
@@ -1,12 +1,3 @@
--- Patch 1.10.2 (2006-05-02): This ability will no longer stack with [Arcane Power].
--- If you gain both at the same time, Arcane Power will take precedence.
-
--- If you have Power Infusion Aura and cast Arcane Power, Power Infusion should be replaced by Arcane Power.
--- But if you have Arcane Power and try to cast Power Infusion, you should get msg "A more powerful spell is active"
--- Arcane Power and Power Infusion don't have the same effect, so stack_rule 4 (SPELL_GROUP_STACK_RULE_EXCLUSIVE_HIGHEST) isn't valid.
--- So, I put stack_rule 1 (SPELL_GROUP_STACK_RULE_EXCLUSIVE) for prevent stack the two auras.
--- And SpellScript for get error msg when you cast Power Infusion when target have Arcane Power Aura.
-
 SET @GROUP_ID := 1122; -- Change with the proper ID if this is in use
```
Comment (Aokromes) [inline sql/updates/world/3.3.5/9999_99_99_99_world_335.sql]: add -- before, incase someone mess eol on previous file xd

Comment (Jildor) [reply sql/updates/world/3.3.5/9999_99_99_99_world_335.sql]: done ;)

File: sql/updates/world/3.3.5/9999_99_99_99_world_335.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
+--
+SET @GROUP_ID := 1122; -- Change with the proper ID if this is in use
```
Comment (jackpoz) [inline sql/updates/world/3.3.5/9999_99_99_99_world_335.sql]: could you update this after https://github.com/TrinityCore/TrinityCore/blob/3.3.5/sql/updates/world/3.3.5/2019_07_24_00_world.sql got merged ?

**Review by jackpoz — CHANGES_REQUESTED** [review]

File: sql/updates/world/3.3.5/9999_99_99_99_world_335.sql (hunk 2 above)
Comment (Jildor) [reply sql/updates/world/3.3.5/9999_99_99_99_world_335.sql]: sure ;)

Comment (Jildor) [reply sql/updates/world/3.3.5/9999_99_99_99_world_335.sql]: done ;)

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Pull request merged**